tests/auto/finddebugsym/tst_finddebugsym.cpp

```diff
@@ -0,0 +1,104 @@
+/****************************************************************************
+**
+** Copyright (C) 2017 Klarälvdalens Datakonsult AB, a KDAB Group company, info@kdab.com, author Milian Wolff
+*<milian.wolff@kdab.com>
+** Contact: http://www.qt.io/licensing/
+**
+** This file is part of the Qt Enterprise Perf Profiler Add-on.
+**
+** GNU General Public License Usage
+** This file may be used under the terms of the GNU General Public License
+** version 3 as published by the Free Software Foundation and appearing in
+** the file LICENSE.GPLv3 included in the packaging of this file. Please
+** review the following information to ensure the GNU General Public License
+** requirements will be met: https://www.gnu.org/licenses/gpl.html.
+**
+** If you have questions regarding the use of this file, please use
+** contact form at http://www.qt.io/contact-us
+**
+****************************************************************************/
+
+#include "perfkallsyms.h"
+
+#include <QDebug>
+#include <QObject>
+#include <QTemporaryDir>
+#include <QTest>
+
+#include "perfsymboltable.h"
+
+QFileInfo findDebugInfoFile(const QString& root, const QString& file, const QString& debugLinkString);
+
+class TestFindDebugSymbols : public QObject
+{
+    Q_OBJECT
+private slots:
+    void initTestCase()
+    {
+        tempDir = new QTemporaryDir();
+
+        const auto files = {
+            QStringLiteral("/usr/bin/python3.8"),
+            QStringLiteral("/usr/bin/.debug/096cdc8214a805dca8d174fe072684b0f21645.debug"),
+            QStringLiteral("/usr/lib/libm.so"),
+            QStringLiteral("/usr/lib/debug/libm.so"),
+            QStringLiteral("/usr/lib/x64/libc.so"),
+            QStringLiteral("/usr/lib/debug/lib/x64/libc.so"),
+            QStringLiteral("/usr/lib/libqt.so"),
+        };
+
+        QDir dir(tempDir->path());
+        for (const auto& file : files) {
+            auto path = QFileInfo(tempDir->path() + file).path();
+            dir.mkpath(path);
+            QFile f(tempDir->path() + file);
+            f.open(QIODevice::WriteOnly);
+            f.write(file.toUtf8());
+            f.close();
+        }
+    }
+
+    void testDotDebug()
```

Comment:
all of these tests should be data driven, cf. QTest::addColumn/addRow and QFETCH

then it should just be a matter of feeding the data to be passed to the findDebugInfoFile call as columns in the various rows. and the test itself just QCOMPARES the result against the expected value you also put into a column value